lib/segment/src/vector_storage/simple_sparse_vector_storage.rs

```diff
@@ -105,15 +100,16 @@ impl SimpleSparseVectorStorage {
         vector: Option<&SparseVector>,
     ) -> OperationResult<()> {
         // Write vector state to buffer record
-        let record = &mut self.update_buffer;
-        record.deleted = deleted;
+        let record = StoredSparseVector {
+            deleted,
+            vector: vector.cloned().unwrap_or_default(),
+        };
```

Comment:
This change is not related to the issue being fixed?

(not an objection to include it, just trying to follow)

Reply:
yes, not directly related (sorry)